spec/02-integration/04-admin_api/15-off_spec.lua

```diff
@@ -2323,6 +2362,76 @@ R6InCcH2Wh8wSeY5AuDXvu2tv9g/PW9wIJmPuKSHMA==
     }, post_config(input))
   end)
 
+  it("flatten_errors when conflicting inputs", function()
+    local conflicting_input = {
+      _format_version = "3.0",
+      consumers = {
```

Comment:
Hmm :thinking: I don't know if it's possible to properly flatten errors in this case, because the result we get from the declarative validator is ambiguous. Consider this input:

```lua
{
  _format_version = "3.0",
  consumers = {
    {
      username = "consumer-01",
      basicauth_credentials = {
        { username = "user-01", password = "pwd" },
        { username = "user-11", password = "pwd" },
        { username = "user-02", password = "pwd" },
        { username = "user-11", password = "pwd" },
      }
    },
    {
      username = "consumer-02",
      basicauth_credentials = {
        { username = "user-99", password = "pwd" },
      }
    },
    {
      username = "consumer-03",
      basicauth_credentials = {
        { username = "user-01", password = "pwd" },
        { username = "user-33", password = "pwd" },
        { username = "user-02", password = "pwd" },
        { username = "user-33", password = "pwd" },
      }
    }
  }
}
```

It has the following duplicates:

1. `consumers[1].basicauth_credentials[4]` (`user-11`)
2. `consumers[3].basicauth_credentials[1]` (`user-01`)
3. `consumers[3].basicauth_credentials[3]` (`user-02`)
4. `consumers[3].basicauth_credentials[4]` (`user-33`)

The error table we receive from `:parse_table()` looks like this:

```lua
{
  basicauth_credentials = {
    [1] = "uniqueness violation: 'basicauth_credentials' entity with username set to 'user-01' already declared",
    [3] = "uniqueness violation: 'basicauth_credentials' entity with username set to 'user-02' already declared",
    [4] = "uniqueness violation: 'basicauth_credentials' entity with username set to 'user-33' already declared",
  },
}
```

1. The declarative code actually overwrote the error in `consumers[1].basicauth_credentials[4]` with the error in `consumers[3].basicauth_credentials[4]` (presumably because they both happened at index `4`)
2. We've lost all context about which consumer the error corresponds to

In summary, I don't think this can be fixed in the error-flattening code alone. A fix within `kong.db.schema.others.declarative_config` must be made first.

Reply:
@flrgh 
I try to add your inputs to test case, it can get 4 results.

Is this what you need?